### PR TITLE
Admin Bar: Replace the wp-admin url with the calypso ones for the default interface

### PR DESCRIPTION
**Pull request by arthur791004**
Fixes https://github.com/Automattic/dotcom-forge/issues/8333

## Proposed changes:

* The follow-up of https://github.com/Automattic/jetpack/pull/38399 to replace the wp-admin URL with the calypso ones for the default interface.
* The concept is to extend the `WP_Admin_Bar` and override the `add_node` function, so we can check whether the URL needs to point to Calypso or not
* ~~Just found the loading order changed, so fix the admin bar on default atomic sites by https://github.com/Automattic/jetpack/pull/38377/commits/2c8201793c0fa7d4863a18eb252aac1437b85574~~
* ~~Fixes the Menu toggle (mobile) on default atomic sites~~

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to /wp-admin on your site with the `default` admin interface 
* Make sure the URLs are replaced with the Calypso ones as much as possible
* Go to  Jetpack -> Settings -> Writing -> Custom Content Types, and enable both Testimonials and Portfolios
* Make sure the following URLs point to Calypso
  * +New -> Project (Portfolios)
  * +New -> Testimonials
* Here are URLs that keep the same
  * Widgets - /wp-admin/widgets.php
  * Menu - /wp-admin/nav-menus.php
  * Jetpack Stats on atomic frontend page - /wp-admin/admin.php?page=stats&noheader&proxy&chart=admin-bar-hours-scale
  * Edit comment icon on p2 - /wp-admin/edit-comments.php
  * Customize - /wp-admin/customize.php?url=%your_site%
  * Manage Site - https://wordpress.com/wp-admin/network/site-info.php?id=%your_site_id%
* Check the Menu toggle on Default atomic sites with mobile view.

### CODE REVIEW

**Review by fushar — COMMENTED** [review]
We also need the following:

- `wp-admin/post-new.php?post_type=jetpack-testimonial` => `https://wordpress.com/types/jetpack-testimonial/%home_url%`
- `wp-admin/post-new.php?post_type=jetpack-portfolio` => `https://wordpress.com/types/jetpack-portfolio/%home_url%`

**Comment by arthur791004** [comment]
> wp-admin/post-new.php?post_type=jetpack-testimonial => https://wordpress.com/types/jetpack-testimonial/%home_url%
> wp-admin/post-new.php?post_type=jetpack-portfolio => https://wordpress.com/types/jetpack-portfolio/%home_url%

Where are these 2 items?

**Comment by fushar** [comment]
> Where are these 2 items?

Ah sorry, you need to enable them on Jetpack -> Settings -> Writing -> Custom Content Types first.

image

File: projects/packages/masterbar/src/class-main.php [hunk 1]
```diff
@@ -29,13 +29,13 @@ public static function init() {
...
 
-		if ( ( defined( 'WPCOM_ADMIN_BAR_UNIFICATION' ) && WPCOM_ADMIN_BAR_UNIFICATION ) || get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
 			return;
 		}
 
 		$host = new Host();
 
-		if ( ! $host->is_wpcom_simple() ) {
+		if ( ! $host->is_wpcom_simple() && ! ( defined( 'WPCOM_ADMIN_BAR_UNIFICATION' ) && WPCOM_ADMIN_BAR_UNIFICATION ) ) {
```
Comment (arthur791004) [inline projects/packages/masterbar/src/class-main.php]: I'm unsure but it seems to be okay to just disable Masterbar if `WPCOM_ADMIN_BAR_UNIFICATION` is enabled

File: projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php [hunk 2]
```diff
@@ -140,6 +134,13 @@ public static function load_wpcom_user_features() {
 		require_once __DIR__ . '/features/wpcom-sidebar-notice/wpcom-sidebar-notice.php';
 		require_once __DIR__ . '/features/wpcom-site-management-widget/class-wpcom-site-management-widget.php';
 		require_once __DIR__ . '/features/wpcom-themes/wpcom-themes.php';
+
+		// Only load the Calypsoify and Masterbar features on WoA sites.
+		if ( class_exists( '\Automattic\Jetpack\Status\Host' ) && ( new \Automattic\Jetpack\Status\Host() )->is_woa_site() ) {
+			\Automattic\Jetpack\Calypsoify\Jetpack_Calypsoify::get_instance();
+			// This is temporary. After we cleanup Masterbar on WPCOM we should load Masterbar for Simple sites too.
+			\Automattic\Jetpack\Masterbar\Main::init();
```
Comment (arthur791004) [inline projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php]: I assume these 2 are wpcom user features, right 🤔

Comment (fushar) [reply projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php]: Yes, but I didn't want to make too many changes in the last PR (https://github.com/Automattic/jetpack/pull/38364)

Are this changes really necessary here?

Comment (arthur791004) [reply projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php]: At least it should be called after the flag defined.

Comment (fushar) [reply projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php]: Ohhhhhh.... I seee....

**Review by fushar — APPROVED** [review]
Tested and I believe it works.

**Review by fushar — DISMISSED** [review]
This causes fatal error in WPCOM 🥲 

```
Fatal error: Cannot redeclare wpcom_custom_admin_bar_class() (previously declared in <redacted>/admin-bar.php:109)
```

**Review by fushar — DISMISSED** [review]
Retested and it looks good on Atomic/Simple, Default/Classic, admin/frontend.

**Comment by fushar** [comment]
@lupus2k The W menu is not working on:

- Atomic,
- Default,
- on a wp-admin page,
- in mobile viewport.

I think I'm gonna revert all changes not related to `+ New` button from this PR and move them to another one. It's getting confusing now...

**Comment by fushar** [comment]
@lupus2k The W menu is not working on:

- Atomic,
- Default,
- on site frontend,
- in mobile viewport.

I think I'm gonna revert all changes not related to `+ New` button from this PR and move them to another one. It's getting confusing now...

**Review by fushar — APPROVED** [review]

**Comment by fushar** [comment]
@lupus2k , @arthur791004 : I've ported the masterbar unification fix to a new PR: https://github.com/Automattic/jetpack/pull/38419.

**Comment by Addison-Stavlo** [comment]
It looks like this added file threw fatals when jetpack released today. Wpcomsh is pinned at 4.0.0 now (which includes the jetpack-mu version it loads) until resolved - p1721840686641099-slack-C06DN6QQVAQ and p1721843060452409-slack-C06DN6QQVAQ

**Pull request merged**